Find a functionally equivalent code:com.android.internal.widget.LockPatternUtils.isLockPasswordEnabled:COMMENT
Method Modifier: public      internal    
Comment:/**
 * @return Whether the lock password is enabled, or if it is set as a backup for biometric weak
 */

@@ -1,7 +1,7 @@
 {
     long mode = getLong(PASSWORD_TYPE_KEY, 0);
     long backupMode = getLong(PASSWORD_TYPE_ALTERNATE_KEY, 0);
-    final boolean passwordEnabled = mode == DevicePolicyManager.PASSWORD_QUALITY_ALPHABETIC || mode == DevicePolicyManager.PASSWORD_QUALITY_NUMERIC || mode == DevicePolicyManager.PASSWORD_QUALITY_ALPHANUMERIC || mode == DevicePolicyManager.PASSWORD_QUALITY_COMPLEX;
-    final boolean backupEnabled = backupMode == DevicePolicyManager.PASSWORD_QUALITY_ALPHABETIC || backupMode == DevicePolicyManager.PASSWORD_QUALITY_NUMERIC || backupMode == DevicePolicyManager.PASSWORD_QUALITY_ALPHANUMERIC || backupMode == DevicePolicyManager.PASSWORD_QUALITY_COMPLEX;
+    final boolean passwordEnabled = mode == DevicePolicyManager.PASSWORD_QUALITY_ALPHABETIC || mode == DevicePolicyManager.PASSWORD_QUALITY_NUMERIC || mode == DevicePolicyManager.PASSWORD_QUALITY_NUMERIC_COMPLEX || mode == DevicePolicyManager.PASSWORD_QUALITY_ALPHANUMERIC || mode == DevicePolicyManager.PASSWORD_QUALITY_COMPLEX;
+    final boolean backupEnabled = backupMode == DevicePolicyManager.PASSWORD_QUALITY_ALPHABETIC || backupMode == DevicePolicyManager.PASSWORD_QUALITY_NUMERIC || backupMode == DevicePolicyManager.PASSWORD_QUALITY_NUMERIC_COMPLEX || backupMode == DevicePolicyManager.PASSWORD_QUALITY_ALPHANUMERIC || backupMode == DevicePolicyManager.PASSWORD_QUALITY_COMPLEX;
     return savedPasswordExists() && (passwordEnabled || (usingBiometricWeak() && backupEnabled));
 }

